common/lib/capa/capa/responsetypes.py

```diff
@@ -851,23 +851,21 @@ class ChoiceResponse(LoncapaResponse):
     def setup_response(self):
         self.assign_choice_names()
 
-        correct_xml = self.xml.xpath(
-            '//*[@id=$id]//choice[@correct="true"]',
-            id=self.xml.get('id')
-        )
+        self.correct_choices = set()
+        self.incorrect_choices = set()
 
-        self.correct_choices = set([
-            choice.get('name') for choice in correct_xml
-        ])
+        cxml = self.xml.xpath('//*[@id=$id]//choice', id=self.xml.get('id'))
```

Comment:
@pomegranited This magic incantation: `self.xml.xpath('//*[@id=$id]//choice'` is used in three different places in this file. It might make sense to factor it out into a separate method.

Also, rename `cxml` to something like `choice_xml`, perhaps?